autoload.php

```diff
@@ -50,6 +50,47 @@ function( $object ) {
 		}
 	}
 );
+/**
+ * Autoloader for components in a theme.
+ */
+spl_autoload_register(
+	function( $class ) {
+
+		// Filter to define the namespace.
+		$theme_component_namespace = apply_filters( 'wp_components_theme_components_namespace', '' );
```

Comment:
Not sure if this would make sense here as this loader appears to for things outside of the `WP_Components` namespace, but instead of an empty string default we could use `__NAMESPACE__`.